Added functions, documentation, and test

mfhstdt opened the pull request:
Added functions for plotting, incorporated plots in shiny app, added documentation and test of plot_lifeSat().

You can find the plot_lifeSat() function and its documentation in "./R/compare_life_satisfaction.R". If you want to test it, open the EuroCity R Project, run `devtools::load_all()` and then `?plot_lifeSat`

The tests for plot_lifeSat() are stored in "./R/tests/testthat/test-plot_lifeSat.R". You can execute them by running: 
`testthat::test_file("tests/testthat/test-plot_lifeSat.R")`

mfhstdt committed:
added functions, documentation, and test

Added functions for plotting, incorporated plots in shiny app, added documentation and test of plot_lifeSat()

mfhstdt committed:
Added functions for price data

mfhstdt committed:
Changed shiny ui & server

lisettedeputter commented on R/compare_life_satisfaction.R:
```diff
@@ -0,0 +1,106 @@
+# SCRIPT TO COMPARE LIFE SATISFACTION ON COUNTRY LEVEL
```
I would change the name of this .R file to "plot_lifeSat" because I first thought that this was a different function, and you did the documentation and testing on two different functions. By calling this .R file "plot_lifeSat.R" you make it more clear.

lisettedeputter replied:
I see now that you had specified this in a comment, so this is mostly my fault. But I do still think you have to save all your functions in separate files, and give the files the name of the function it contains. (Though I am not 100% sure this is necessary, so by all means ignore this if this is not the case!)

lisettedeputter commented on R/compare_living_expenses.R:
```diff
@@ -0,0 +1,44 @@
+# SCRIPT TO COMPARE LIFE SATISFACTION ON COUNTRY LEVEL
```
No documentation of this function?

lisettedeputter commented on tests/testthat/test-plot_lifeSat.R:
```diff
@@ -0,0 +1,24 @@
+
```
Good tests! Because: they test the actual different outcomes the function should give in different scenario's and checks the labels. For me though the third test fails, the line 
expect_identical(p$labels$title, "Self-reported life satisfaction in Netherlands and Greece (2003 - 2021)") 
fails because p$labels$title is "Self-reported life satisfaction" and not what you want it to be. I don't know exactly why, but maybe you can do something with this?

mfhstdt replied:
Good that you spot this! I changed the function a few days ago and forgot to adapt the test. Thanks for pointing this out! :)

lisettedeputter reviewed (commented):
Nice Project! Looks very interesting! The documentation is clear and tests are good. Everything is also clear for outsiders, e.g., good use of comments in the code so it is easily understandable! Only thing: missing documentation for the other functions than "plot_lifeSat" but I think we did only have to do one function for now, (but to make it complete add them for the other usable functions too).

lisettedeputter commented on tests/testthat/test-plot_lifeSat.R:
```diff
@@ -0,0 +1,24 @@
+  p <- plot_lifeSat("Netherlands", "Greece")
+  expect_identical(p$labels$y, "Self-reported life satisfaction")
+  expect_identical(p$labels$x, "Year")
+  expect_identical(p$labels$title, "Self-reported life satisfaction in Netherlands and Greece (2003 - 2021)")
```
So this line fails for me. What R says: 
```
✔ | F W S  OK | Context
✖ | 1       5 | plot_lifeSat [0.2s]                                                    
─────────────────────────────────────────────────────────────────────────
Failure (test-plot_lifeSat.R:20): Labels, title, and caption correct
p$labels$title (`actual`) not identical to "Self-reported life satisfaction in Netherlands and Greece (2003 - 2021)" (`expected`).
`actual`:   "Self-reported life satisfaction"                                        
`expected`: "Self-reported life satisfaction in Netherlands and Greece (2003 - 2021)"
```

rebekka-cebulla commented on R/compare_life_satisfaction.R:
```diff
@@ -0,0 +1,106 @@
+#' Plot the trajectories of self-reported life satisfaction of two chosen European
+#' countries between 2003 and 2021
+#'
+#' @description
```
I think this documentation is very useful and clear for the reader!

rebekka-cebulla reviewed (commented):
There are some functions without documentation it would be nice to add those but other than that I think its really nice! I also like the comments you made inside your code to make things clear for others or you in the future!

mfhstdt committed:
imports and external data

Added external data & documented it; added informative errors to functions; replaced library() and load() calls with imports
